android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -385,31 +394,28 @@ protected int getThemeResourceId(@NonNull String theme)
     return super.getThemeResourceId(theme);
   }
 
-  private void replaceMapWithPlaceholder()
+  @Override
+  public void onDisplayChangedToCar(@NonNull OnTaskFinishedCallback onTaskFinishedCallback)
   {
-    disableControls();
-    if (mMapFragment != null)
-    {
-      getSupportFragmentManager().beginTransaction().remove(mMapFragment).commitNowAllowingStateLoss();
-      mMapFragment = null;
-    }
-    getSupportFragmentManager().beginTransaction().replace(R.id.map_fragment_container, new MapPlaceholderFragment()).commitNowAllowingStateLoss();
+    mRemoveDisplayListener = false;
+    startActivity(new Intent(this, MapPlaceholderActivity.class));
+    Objects.requireNonNull(mMapFragment).notifyOnSurfaceDestroyed(onTaskFinishedCallback::onTaskFinished);
+    finish();
   }
 
-  private void replacePlaceholderWithMap()
+  @Override
+  public void onConfigurationChanged(@NonNull Configuration newConfig)
   {
-    ThemeSwitcher.INSTANCE.restart(false);
-    initMap(false /* isLaunchByDeepLink */, true /* force */);
-    enableControls();
-  }
+    super.onConfigurationChanged(newConfig);
 
-  private void mapDisplayChanged(@NonNull final DisplayType newDisplayType)
-  {
-    Logger.d(TAG);
-    if (newDisplayType == DisplayType.Device)
-      replacePlaceholderWithMap();
-    else
-      replaceMapWithPlaceholder();
+    final int newUiMode = newConfig.uiMode & Configuration.UI_MODE_TYPE_MASK;
+    final boolean newUiModeIsCarConnected = newUiMode == Configuration.UI_MODE_TYPE_CAR;
+    final boolean newUiModeIsCarDisconnected = mLastUiMode == Configuration.UI_MODE_TYPE_CAR && newUiMode == Configuration.UI_MODE_TYPE_NORMAL;
+    mLastUiMode = newUiMode;
+
+    if (newUiModeIsCarConnected || newUiModeIsCarDisconnected)
+      return;
+    recreate();
```

Comment:
When AA connected/disconnected nothing happens.

`onConfigurationChanged` can only handle the case when the device is connected to the AA. It can't notify the application when the user runs our app on the AA.

This code only stops activity recreations when user connects or disconnets AA. It also provides default behavior for another UI changes (dock, watch and night mode) - activity recreation.